src/main/java/thrift/logic/parser/ThriftParser.java

```diff
@@ -104,10 +104,20 @@ public Command parseCommand(String userInput) throws ParseException {
             return new ListCommandParser().parse(arguments);
 
         case RedoCommand.COMMAND_WORD:
-            return new RedoCommand();
+            if (arguments == null || arguments.trim().equals("")) {
+                return new RedoCommand();
+            } else {
+                throw new ParseException(String.format(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        RedoCommand.MESSAGE_USAGE)));
+            }
 
         case UndoCommand.COMMAND_WORD:
-            return new UndoCommand();
+            if (arguments == null || arguments.trim().equals("")) {
+                return new UndoCommand();
+            } else {
+                throw new ParseException(String.format(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        UndoCommand.MESSAGE_USAGE)));
+            }
```

Comment:
Maybe can make a `EmptyArgumentsCommandParser(String MESSAGE_USAGE`)` class which can be used by `redo` `undo` and `exit` since the code are the same (checking for non-args)

Reply:
`NoArgumentsCommandParser` needs to implement `Parser` as it performs parsing. 

So, I changed to this instead:
```java
case RedoCommand.COMMAND_WORD:
            return new NoArgumentsCommandParser(commandWord, RedoCommand.MESSAGE_USAGE).parse(arguments);
```